Uplift/Views/Cells/FacilitiesDropdownCollectionViewCell.swift

```diff
@@ -0,0 +1,168 @@
+    var height: CGFloat!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setupDropdownView()
+        setupConstraints()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func configure(for facility: Facility) {
+        self.facility = facility
+        setupDropdownView()
+        setupConstraints()
+    }
+    
+    func setupDropdownView() {
+        headerView = setupHeaderView(headerString: facility.name.uppercased())
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: StretchyHeaderLayout())
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        //TODO: register facility cell classes once they're finished
+        
+        collectionViewHeight = 0 //TODO: once cell classes are finished, sum up their heights to get the total collection view height
+        
+        dropdownView = DropdownView(delegate: self,
+                                    headerView: headerView,
+                                    headerViewHeight: headerViewHeight,
+                                    contentView: collectionView,
+                                    contentViewHeight: collectionViewHeight)
+    }
+    
+    func setupHeaderView(headerString: String) -> UIView {
+        var headerImage: UIImage
+        switch headerString {
+        case "EQUIPMENT", "FITNESS CENTER":
+            headerImage = UIImage(named: ImageNames.equipment)!
+        case "GYMNASIUM":
+            headerImage = UIImage(named: ImageNames.basketball)!
+        case "SWIMMING POOL":
+            headerImage = UIImage(named: ImageNames.pool)!
+        case "BOWLING LANES":
+            headerImage = UIImage(named: ImageNames.bowling)!
+        default:
+            headerImage = UIImage(named: ImageNames.misc)!
+        }
+        
+        headerView = UIView()
+        
+        headerImageView = UIImageView(image: headerImage)
+        headerView.addSubview(headerImageView)
+        
+        headerNameLabel = UILabel()
+        headerNameLabel.text = headerString
+        headerNameLabel.font = ._16MontserratRegular
+        headerNameLabel.textColor = .black
+        headerNameLabel.textAlignment = .left
+        headerView.addSubview(headerImageView)
+        
+        let facilityOpen = checkFacilityOpen()
+        headerOpenLabel = UILabel()
+        headerOpenLabel.font = ._16MontserratRegular
+        headerOpenLabel.text = facilityOpen ? "Open" : "Closed"
+        headerOpenLabel.textColor = facilityOpen ? .accentOpen : .accentClosed
+        headerOpenLabel.textAlignment = .center
+        headerView.addSubview(headerOpenLabel)
+        
+        return headerView
+    }
+    
+    func checkFacilityOpen() -> Bool {
+        for detail in facility.details {
+            if detail.times.count != 0 {
+                let today = Date().getIntegerDayOfWeekToday()
```

Comment:
move this before the for loop since it doesnt change and rename it to `dayOfWeek`